test/e2e_node/log_path_test.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.KubeDescribe("ContainerLogPath", func() {
+	f := framework.NewDefaultFramework("kubelet-container-log-path")
+	Describe("Pod containers", func() {
+		Context("with log printed to stdout", func() {
+			var ns, logDir string
+			BeforeEach(func() {
+				ns = f.Namespace.Name
+				logDir = "/var/log/containers/"
+			})
+			It("Correct log path on the node should have been created", func() {
+				podName := "log-pod" + string(uuid.NewUUID())
+				logContName := "log-container" + string(uuid.NewUUID())
+				// The expected container's log file name on host is:
+				// <pod_name>_<pod_namespace>_<container_name>-<container_id>.log
+				logPathPrefix := podName + "_" + ns + "_" + logContName + "-"
+
+				pod := &api.Pod{
+					ObjectMeta: api.ObjectMeta{
+						Name:      podName,
+						Namespace: ns,
+					},
+					Spec: api.PodSpec{
+						Containers: []api.Container{
+							{
+								Image:   "gcr.io/google_containers/busybox:1.24",
+								Name:    logContName,
+								Command: []string{"sh", "-c", "while true; do echo \"$(hostname)\"; sleep 1; done"},
+							},
+						},
+					},
+				}
+
+				pod, err := f.Client.Pods(ns).Create(pod)
+				if err != nil {
+					framework.Failf("unable to create pod %v: %v", podName, err)
+				}
+
+				err = f.WaitForPodRunning(podName)
+				Expect(err).NotTo(HaveOccurred(), "Failed waiting for pod %s to enter running state", podName)
+
+				files, err := ioutil.ReadDir(logDir)
```

Comment:
I've tried like that before but it won't work, and I think I should blame docker this time.

If I do `ln ~/check.sh ~/check.sh.link`, and then `docker run -it -v ~:/test/ busybox cat /test/check.sh.link`, it works.

But if I create a pod in k8s, it will fail:
```
docker run -it -v /var/log/containers:/test busybox
/ # ls test/
logging-sidecar-example_default_POD-379fee5281ce4136b56ddb9d5a0d1c6e373dd80f06897a2b735313870648df4c.log
logging-sidecar-example_default_synthetic-logger-2e9238c72824686f121abb69ead28b23d387604627bc0a7b3112a00f3d0c3e6a.log
```
I cannot read this file anyhow:
```
/ # cat logging-sidecar-example_default_synthetic-logger-2e9238c72824686f121abb69ead28b23d387604627bc0a7b3112a00f3d0c3e6a.log
cat: can't open 'logging-sidecar-example_default_synthetic-logger-2e9238c72824686f121abb69ead28b23d387604627bc0a7b3112a00f3d0c3e6a.log': No such file or directory
```